src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1092,6 +1092,12 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
             ImmutableArray<Location> memberLocations,
             ImmutableArray<bool> memberIsReadOnly);
 
+        /// <summary>
+        /// Returns true if there is an implicit (C#) or widening (VB) conversion from
+        /// <paramref name="fromType"/> to <paramref name="toType"/>.
+        /// </summary>
+        public abstract bool IsAssignableTo(ITypeSymbol fromType, ITypeSymbol toType);
```

Comment:
>IsAssignableTo [](start = 29, length = 14)

The name sounds very similar to a well known API ```Type.IsAssignableFrom```, but according to the doc comment doesn't quite match the behavior. This could be a source of big confusion for consumers and as a result a source of misuse of the API.

Reply:
I like `IsAssignableTo`, short & clear.

Reply:
Also the fact that in VB a widening conversion is not a requirement for things to be assignable should be taken into consideration when choosing the name of this API

Reply:
@AlekseyTs can you give some examples?  it would be super helpful for getting everyone understanding this space better.  Thanks!

Reply:
I don't know what "safe" means. `IsAssignableTo` is the name used everywhere I've seen people need to do it themselves, so there appears to be consensus around the current name.

I understand that there are `Option` settings in VB where assignments have much looser rules (e.g. `Option Strict Off`). That would allow an assignment from, for example, `IEnumerable<char>` to `String` - it throws an exception if the conversion fails, I suppose.

Reply:
> I don't know what "safe" means. 

I actually doubt that. In any case, that was just an example, feel free to come up with a different word that accurately describes characteristics of an implicit C# conversion. I assume this is what this API is after. Here is how language specification puts it: "The pre-defined implicit conversions always succeed and never cause exceptions to be thrown. Properly designed user-defined implicit conversions should exhibit these characteristics as well."

> IsAssignableTo is the name used everywhere I've seen people 
need to do it themselves, so there appears to be consensus around the current name.

Not sure what consensus you have in mind and how wide was the scope of things you looked at. Clearly there are existing APIs in .NET framework that use "IsAssignable" term and use different meaning behind it. Have you looked at this API and their usage?
I personally think that, if API is about conversion, the name should reflect that, rather than focusing on a very narrow use case that happened to be about an assignment scenario.

Reply:
>  "The pre-defined implicit conversions always succeed and never cause exceptions to be thrown. Properly designed user-defined implicit conversions should exhibit these characteristics as well."

interesting.  Based on that, perhaps a good name would be ```bool ConversionWillAlwaysSucceed(ITypeSymbol fromType, ITypeSymbol toType)```?  I would at least get a very good sense of what it means.  And it would serve as a template for what we might need to agument in the future if IsImplicit is insufficient.

Reply:
`IsImplicitConversion(ITypeSymbol fromType, ITypeSymbol toType)` is shorter if so.

Reply:
`HasImplicitConversion` or `CanBeImplicitlyConverted`? This is the primary check being performed by the analyzers in #9461

Reply:
What does that *mean* though?  In particular, what do you define "implicit" as.  And i'm with Aleksey, what does this mean in VB?

Reply:
I'm willing to look through the links, but it would be useful to have primary use cases where the feature can describe precisely the sort of question it is trying to ask (in a non-self referential fashion).  i.e. we shouldn't say "teh feature is trying to know if it's an implicit conversion".  We should say "teh feature is trying to decide if applying the conversion at runtime would succeed or not".

Reply:
For example, VB says this: "Implicit conversions occur without any special syntax".  Is it true then that .isWidening is the right impl to answer that question?  To me, it looks like no.  For example:

> If permissive semantics are being used, all widening and narrowing conversions (in other words, all conversions) may occur implicitly.

Reply:
There are places in C# where non-implicit conversions occur implicitly (e.g. in a foreach loop). I'll make sure the doc comments are clear.

---
In reply to: [186879526](https://github.com/dotnet/roslyn/pull/26719#discussion_r186879526) [](ancestors = 186879526)

Reply:
Since we now have a team for deciding public CodeAnalysis API changes, I'll submit this question to that group.

---
In reply to: [186923887](https://github.com/dotnet/roslyn/pull/26719#discussion_r186923887) [](ancestors = 186923887,186879526)